chain/blocksync/blocksync_client.go

```diff
@@ -101,8 +101,13 @@ func (bs *BlockSync) GetBlocks(ctx context.Context, tsk types.TipSetKey, count i
 		}
 
 		if res.Status == 0 {
+			resp, err := bs.processBlocksResponse(req, res)
+			if err != nil {
+				return nil, xerrors.Errorf("success response from peer failed to process: %w", err)
+			}
 			bs.syncPeers.logGlobalSuccess(time.Since(start))
-			return bs.processBlocksResponse(req, res)
+			bs.host.ConnManager().TagPeer(p, "bsync", 2)
```

Comment:
FYI, DHT peers will have a higher priority (`5`, or `5 + log2(xor distance)` when you next update).

Reply:
Also note: Bitswap adds a pretty significant weight by default (even for peers _downloading_ from us). We may need to tune that.